salsa-2022-tests/tests/hello_world.rs

```diff
@@ -44,6 +44,10 @@ impl salsa::Database for Database {
     fn salsa_runtime(&self) -> &salsa::Runtime {
         self.storage.runtime()
     }
+
+    fn salsa_runtime_mut(&mut self) -> &mut salsa::Runtime {
+        self.storage.runtime_mut()
+    }
```

Comment:
Hmm, I am not thrilled about requiring everyone to add this method just so that they can make synthetic writes!

I think I would rather add the "synthetic write" method to `HasJarsDyn` or some plumbing trait, or else add the `salsa_runtime_mut` method into that trait.

(For that matter, why do we make everyone add the `salsa_runtime` method...? I don't think there's a good reason, the code in `salsa::db` already assumes it can find the `storage` field...)

Reply:
@nikomatsakis  I added `synthetic_write` to `HasJarsDyn`, but using `HasJarsDyn` exposes many APIs that users don't care about, such as `maybe_changed_after`, `remote_stale_output`.

Reply:
> Would it be a good idea to add synthetic_write to the salsa::Database trait and provide a default implementation for it?

I think that makes sense, yes.